### ISSUE TITLE
delete resource doesn't work

### ISSUE THREAD

**Issue by danielcoam** [beda-software/fhir-py#71]
I can't make the `delete()` resource function work

**This is my remove-patient function**

```python
client = AsyncFHIRClient('https://fhir-server', authorization='Bearer my-token')

async def remove_patient(patient_id):
    patients = client.resources('Patient')
    patient = await patients.search(_id=patient_id).get()
    await patient.delete()
```

**The Error I get from the FHIR server:**
```js
"description": "Invalid JSON payload received. Unknown name \"_format\": Cannot bind query parameter. Field 'format' could not be found in request message."
```

It looks like the problem is here : https://github.com/beda-software/fhir-py/blob/master/fhirpy/base/lib.py#L90
When adding the `_format=json` to the `DELETE` request
```python
params = params or {}
params['_format'] = 'json' # The Problem
return f'{self.url}/{path.lstrip("/")}?{encode_params(params)}'
```

When I remove this line it works as expected.

Any help? Couldn't find an open issue for that

**Comment by ir4y** [comment]
Hi, @danielcoam 
Thank you for the contribution!

What FHIR server implementation do you use?

**Comment by danielcoam** [comment]
Hi @ir4y
Thanks for the response!

I use FHIR-R4 on [Google Cloud Healthcare API](https://cloud.google.com/healthcare/docs/concepts/fhir)

It looks (http://hl7.org/fhir/http.html#3.1.0.7.1 ) that in very rare cases when there is no support in a _Conditional delete_ there might be an OperationOutcome response.

But the majority of DELETE operations won't support that.

What I did in our project meanwhile is that:
```python
class CustomizedFHIRAsyncClient(AsyncFHIRClient):
	"""
	There is a bug in the library see :https://github.com/beda-software/fhir-py/issues/71
	The only thing we are changing in the `AsyncFHIRClient` is removing the _format=json from DELETE operation
	"""

	async def _do_request(self, method, path, data=None, params=None):
		headers = self._build_request_headers()
		url = self._build_request_url(path, params)
		if method.lower() == 'delete':  # The change from the original function
			url = url.split('_format=json')[0][:-1]
		async with aiohttp.request(
				method, url, json=data, headers=headers
		) as r:
			if 200 <= r.status < 300:
				data = await r.text()
				return json.loads(data, object_hook=AttrDict)

			if r.status == 404 or r.status == 410:
				raise ResourceNotFound(await r.text())

			raise OperationOutcome(await r.text())
```

But it means we are on a fixed version(1.2.0) of the `fhirpy` package and we can't get any updates.

It's a major blocking that I think should be fixed in the library (the only thing that's weird for me is that there isn't an open bug already)

**Comment by ruscoder** [comment]
@danielcoam Now it should work. Please upgrade to 1.2.1
